src/utils/work/seekWorkResponse.js

```diff
@@ -44,56 +53,33 @@ const seekFromAPI = async (
     ...requestProps,
   };
 
-  let result = null;
-
-  if (eventCallback) {
-    io.off(`${experimentId}-${body.name}`);
+  const timeoutPromise = new Promise((resolve, reject) => {
+    const id = setTimeout(() => {
+      clearTimeout(id);
+      reject(new WorkTimeoutError(timeoutDate, request));
+    }, timeout * 1000);
+  });
 
-    io.on(`${experimentId}-${body.name}`, (res) => {
```

Comment:
Why was this removed? What happens when we run a clustering now? Are cell sets updated?

Reply:
This socket is used for 'broadcast' type work requests (I saw the code for broadcast was removed as well)

Reply:
Yes, broadcast was never used, so I removed it. If you check how `seekFromAPI` is called, the parameter which takes `eventCallback` is explicitly set to null.

I moved the event handling to the end of `fetchWork` (https://github.com/hms-dbmi-cellenics/ui/pull/618/files#diff-37e452fd6bfee4dd173ebb5f7be9fb0e91f42bea39f0d203ce497608dab65edbR176-R178), so that the event handlers are still called at the end of the `fetchWork` request.

Reply:
If the broadcast isn't being used anymore I think then there's something in the worker we aren't using anymore too (that handles 'broadcast').

Reply:
Turns out the broadcast is still used to sync cell sets for concurrent users looking at the same experiment. See this comment : https://github.com/hms-dbmi-cellenics/ui/pull/618#discussion_r794847974